ckanext/canada/logic.py

```diff
@@ -185,3 +226,176 @@ def datastore_create_temp_user_table(context):
+
+            if not is_system_process and not current_user:
+                # only remove from the list if a user is not logged in
+                disabled_views_indexes.append(i)
+                continue
+
+            if not is_system_process and current_user != site_user:
+                # only add key/value if not system process
+                view_list[i]['canada_disabled_view'] = True
+    return [v for i, v in enumerate(view_list) if i not in disabled_views_indexes]
+
+
+@chained_action
+def canada_job_list(up_func, context, data_dict):
+    """List enqueued background jobs.
+
+    :param list queues: Queues to list jobs from. If not given then the
+        jobs from all queues are listed.
+
+    :param int limit: Number to limit the list of jobs by.
+
+    :param bool ids_only: Whether to return only a list if job IDs or not.
+
+    :returns: The currently enqueued background jobs.
+    :rtype: list
+
+    Will return the list in the way that RQ workers will execute the jobs.
+    Thus the left most non-empty queue will be emptied firts
+    before the next right non-empty one.
+
+    Canada Fork: fetches some more information from the backend Redis
+    and parses it and maps it into some more dict entries to be
+    used in the custom Job Queue view method and template.
+    """
+    job_list = up_func(context, data_dict)
+
+    for job in job_list:
+        try:
+            job_obj = Job.fetch(job.get('id'))
+            job_args = job_obj.args[0]
+        except (NoSuchJobError, KeyError):
+            continue
+
+        job_title = _(job.get('title', 'Unknown Job'))
+
+        if job_obj.func_name in JOB_MAPPING:
+            rid = JOB_MAPPING[job_obj.func_name]['rid'](job_args)
+            icon = JOB_MAPPING[job_obj.func_name]['icon']
+        else:
+            rid = None
+            job_title = _('Unknown Job')
+            icon = 'fa-circle-o-notch'
+
+        job_info = {}
+        if rid:
+            current_user = get_action('get_site_user')({'ignore_auth': True}, {})['name']
+            if has_request_context():
+                try:
+                    current_user = g.user
+                except (TypeError, AttributeError):
+                    pass
+            try:
+                resource = get_action('resource_show')({'user': current_user}, {'id': rid})
+            except (ObjectNotFound, NotAuthorized):
+                continue
+            job_info = {'name_translated': resource.get('name_translated'),
+                        'resource_id': rid,
+                        'url': h.url_for('dataset_resource.read',
+                                         id=resource.get('package_id'),
+                                         resource_id=rid)}
+
+        job['created_human_readable'] = canada_date_str_to_datetime(job.get('created')) \
+                .replace(tzinfo=utc).astimezone(ottawa_tz) \
+                .strftime('%Y-%m-%d %H:%M:%S %Z')
```

Comment:
All the other APIs return UTC dates so this is unusual. Also why is it called human_readable when it looks almost like ISO8601?

Reply:
Okay, I have left `created` as the normal UTC in the logic method, and just changed the output in the view/template.